pkg/compiler/compiler.go

```diff
@@ -124,21 +148,18 @@ func CompileWithDebugInfo(name string, r io.Reader) ([]byte, *DebugInfo, error)
 
 // CompileAndSave will compile and save the file to disk in the NEF format.
 func CompileAndSave(src string, o *Options) ([]byte, error) {
-	if !strings.HasSuffix(src, ".go") {
-		return nil, fmt.Errorf("%s is not a Go file", src)
-	}
 	o.Outfile = strings.TrimSuffix(o.Outfile, fmt.Sprintf(".%s", fileExt))
 	if len(o.Outfile) == 0 {
-		o.Outfile = strings.TrimSuffix(src, ".go")
+		if strings.HasSuffix(src, ".go") {
+			o.Outfile = strings.TrimSuffix(src, ".go")
+		} else {
+			o.Outfile = "out"
+		}
 	}
 	if len(o.Ext) == 0 {
 		o.Ext = fileExt
 	}
-	b, err := ioutil.ReadFile(src)
-	if err != nil {
-		return nil, err
-	}
-	b, di, err := CompileWithDebugInfo(src, bytes.NewReader(b))
+	b, di, err := CompileWithDebugInfo(src, nil)
```

Comment:
I found one more issue while testing this ability to compile multiple files from main package. We're able to compile non-linked methods, but suppose that we have package `main` with 2 files: `main.go` and `helper.go`:

File `main.go`:
```
package main

func MainFunc() int {
	return HelperFunc()
}
```

File `helper.go`:
```
package main

func HelperFunc() int {
	return 1
}
```

Compiling `main.go` leads to the following error:
```
/home/neospcc/Documents/GitProjects/nspcc-dev/neo-go/examples/main/main.go:4:9: undeclared name: HelperFunc
error while trying to compile smart contract file: couldn't load packages due to errors: main
```

Reply:
This is expected. `go build` behaves similarly:
```
❯ go build main.go
# command-line-arguments
./main.go:4:9: undefined: HelperFunc
```

But I have forgotten to alter `contract compile` CLI to support directories. Will fix.